src/libpgagroal/query_cache.c

```diff
@@ -203,10 +200,26 @@ pgagroal_query_cache_add(struct query_cache* cache, struct hashTable** Table, st
       memset(copy_data->value, 0, data->length);
       memcpy(copy_data->value, data->value, data->length);
       copy_data->length = data->length;
+      int idx = cache->max_elements;
+      pgagroal_log_info("Adding to cache %d", idx);
+
+      cache->cache[idx].key = copy_key;
+      cache->cache[idx].data = copy_data;
+      cache->max_elements = idx + 1;
+      pgagroal_log_info("fromcache to cache %s", cache->cache[idx].key->key);
```

Comment:
More logging...

Reply:
Yeah this is something im testing, will remove after we resolve this issue https://github.com/agroal/pgagroal/discussions/376#discussioncomment-7480504